### PR TITLE
Improve CI syntax and concurrency

### PR DESCRIPTION
**Pull request by qbarrand**
This PR adds the `e2e/e2e` job to the `sro-upstream-ci` [concurrency group](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#concurrency), effectively preventing two jobs from running at the same time against the CI cluster.

### CODE REVIEW

**Comment by qbarrand** [comment]
@pacevedom @dagrayvid please have a look at this. In addition to d1eed60 and 2fc61b2 which fix the CI, this PR makes slight improvements to the CI config and to the way we deploy manifests.

**Comment by yevgeny-shnaidman** [comment]
@qbarrand  are all those commits related to the same PR? Maybe you can just squash them, and present all those changes as one commit?

**Comment by ybettan** [comment]
I just had the same comment on https://github.com/openshift-psap/special-resource-operator/pull/125#issuecomment-962573883 @yevgeny-shnaidman.

The way I see it we should have 1 commit per "logical change" (usually 1 commit per PR or a few if this is a very complex PR)
@qbarrand

**Comment by qbarrand** [comment]
@yevgeny-shnaidman @ybettan Doesn't GitHub squash all commits in a PR when merging? An example would be b853d334c427520b615eb7c2e5eab86d3884ed53.

**Comment by ybettan** [comment]
I think it depends on the configuration of the repo and it can vary from repo to repo.

Even when git auto squash, we may have some non logical commits that we forget to remove.

Anyway, when you look at the code later on with `git blame` it is easier to have 1 or 2 logical changes rather than a lot of small commit messages for each change.

Bottom line, IMO, better a *logical* commit message rather than multiple small *technical* commit messages.

File: Makefile [hunk 1]
```diff
@@ -31,13 +31,22 @@ verify: patch fmt vet
...
+	for c in $(CRDS); do kubectl wait --for condition=established --timeout=60s -f $${c}; done
+	# Apply all the rest
+	kubectl apply --wait -f ./manifests$(SUFFIX)
+
+undeploy-manifests: manifests$(SUFFIX)
+	kubectl delete --wait --ignore-not-found -f ./manifests$(SUFFIX)
+
+# TODO: obsolete targets; remove
+go-deploy-manifests: deploy-manifests
+go-undeploy-manifests: undeploy-manifests
```
Comment (pacevedom) [inline Makefile]: I would do this in a separate PR, this is not strictly related to simplifying and fixing the CI

File: pkg/assets/assets.go [hunk 2]
```diff
@@ -101,5 +101,5 @@ func filePathPatternValid(path string) bool {
 var reState = regexp.MustCompile(`^[0-9]{4}[-_].*\.yaml$`)
 
 func ValidStateName(path string) bool {
-	return reState.MatchString(path)
+	return reState.MatchString(filepath.Base(path))
```
Comment (pacevedom) [inline pkg/assets/assets.go]: Same comment, this is not CI

File: pkg/assets/assets_test.go [hunk 3]
```diff
@@ -14,6 +15,18 @@ func TestValidStateName(t *testing.T) {
...
+			valid: true,
+		},
+		{
+			input: "/0000_test.yaml",
+			valid: true,
+		},
+		{
+			input: "./0000_test.yaml",
+			valid: true,
+		},
```
Comment (pacevedom) [inline pkg/assets/assets_test.go]: Same comment

File: pkg/resource/resource.go [hunk 4]
```diff
@@ -192,7 +192,9 @@ func CreateFromYAML(yamlFile []byte,
 		// We used this for predicate filtering, we're watching a lot of
 		// API Objects we want to ignore all objects that do not have this
 		// label.
-		filter.SetLabel(obj)
+		if err = filter.SetLabel(obj); err != nil {
+			return fmt.Errorf("could not set the label: %w", err)
+		}
```
Comment (pacevedom) [inline pkg/resource/resource.go]: Same comment

File: pkg/resource/resource.go [hunk 5]
```diff
@@ -291,7 +293,9 @@ func CRUD(obj *unstructured.Unstructured, releaseInstalled bool, owner v1.Object
 		logg.Info("Release", "Installed", releaseInstalled)
 		logg.Info("Is", "OneTimer", IsOneTimer(obj))
 
-		hash.Annotate(obj)
+		if err = hash.Annotate(obj); err != nil {
+			return err
+		}
```
Comment (pacevedom) [inline pkg/resource/resource.go]: Same comment

File: pkg/resource/resource.go [hunk 6]
```diff
@@ -328,15 +332,17 @@ func CRUD(obj *unstructured.Unstructured, releaseInstalled bool, owner v1.Object
 	if err != nil {
 		return err
 	}
-	if !equal {
+	if equal {
```
Comment (pacevedom) [inline pkg/resource/resource.go]: This is fixed in #133

File: pkg/resource/resource.go [hunk 7]
```diff
@@ -328,15 +332,17 @@ func CRUD(obj *unstructured.Unstructured, releaseInstalled bool, owner v1.Object
...
 		return nil
 	}
 
 	logg.Info("Found, updating")
 	required := obj.DeepCopy()
 
-	hash.Annotate(required)
+	if err = hash.Annotate(required); err != nil {
+		return err
+	}
```
Comment (pacevedom) [inline pkg/resource/resource.go]: Not related to the CI

**Review by pacevedom — COMMENTED** [review]
I think we should split the PR. Not all changes relate to the CI.

**Comment by dagrayvid** [comment]
> I think it depends on the configuration of the repo and it can vary from repo to repo.

AFAIK, the upstream openshift-psap/special-resource-operator repo is not configured to squash commits. The downstream repo is. If needed we can probably configure that.

**Comment by qbarrand** [comment]
@dagrayvid I think it is. Look at 1b2e32f95872ef68050739de9ec707e68de081db or b1a21cbb539a88b37cecdaf3d9df585e0f195f8f, for instance.

**Comment by dagrayvid** [comment]
> @dagrayvid I think it is. Look at [1b2e32f](https://github.com/openshift-psap/special-resource-operator/commit/1b2e32f95872ef68050739de9ec707e68de081db) or [b1a21cb](https://github.com/openshift-psap/special-resource-operator/commit/b1a21cbb539a88b37cecdaf3d9df585e0f195f8f), for instance.

My apologies, you are correct!

**Comment by qbarrand** [comment]
@yevgeny-shnaidman @ybettan I fixed this PR so that it only has the CI-related changes, including a concurrency setting that should prevent `e2e` jobs from running at the same time against the cluster. PTAL.

File: .github/workflows/e2e.yaml [hunk 8]
```diff
@@ -29,18 +28,22 @@ jobs:
       id: short-sha
       with:
         length: 10
-    - run: sudo apt-get install fuse-overlayfs
```
Comment (dagrayvid) [inline .github/workflows/e2e.yaml:32]: Did you mean to remove this?

Comment (qbarrand) [reply .github/workflows/e2e.yaml:32]: Yes, it is [already installed](https://github.com/openshift-psap/special-resource-operator/runs/4153675034?check_suite_focus=true#step:5:10).

**Comment by qbarrand** [comment]
CI linting failure addressed in #138.

**Review by pacevedom — APPROVED** [review]

**Pull request merged**